.pre-commit-hooks.yaml

```diff
@@ -0,0 +1,6 @@
+- id: pip-compile
+  name: pip-compile
+  description: Automatically compile requirements.
+  entry: pip-compile
+  language: python
+  files: (requirements\.in|setup\.py)$
```

Comment:
I'd suggest `pyproject.toml` and `setup.cfg` as well for completeness

and probably some docs on how to use this / when you might want to use this

Reply:
> I'd suggest `pyproject.toml` and `setup.cfg` as well for completeness

Thanks for noticing! I've missed that out. :(

> and probably some docs on how to use this / when you might want to use this

Could be enough something like https://github.com/asottile/flake8-walrus#as-a-pre-commit-hook?

Reply:
Can we borrow it? :)

Reply:
sure! you may want to suggest `stages: [pre-push]` as well as `pre-commit install --hook-type pre-push` as well

Reply:
lol, you can tell how often I use that feature :upside_down_face: (and I'm the author!)

Reply:
Do we really need to mention the stages? This looks good to me:

```yaml
repos:
  - repo: https://github.com/jazzband/pip-tools
    rev: 4.3.0
    hooks:
      - id: pip-compile
```

Reply:
up to you, not my project :D

Reply:
I've tried this hook as a user on my repositories and it looks `pass_filenames: false` and probably `files` too need to be removed. Otherwise, pre-commit will run something like:

```console
$ pip-compile setup.py setup.cfg
```

which is totally wrong, since setup.cfg is not a requirement file.

I'd let users configure `files` and `args` manually for their special case.

Reply:
if there's no good default, it might make sense to set `files: ^$` + `pass_filenames: false` and document that the consumer must configure `args: [requirements.in]` + `files: ^(...)$` (for example)

Reply:
Addressed in 5f2dea3db93cb5e313b654c0c26750195827b86f